index.js

```diff
@@ -54,10 +58,10 @@ function paginationUpdate(n) {
 
 paginationUpdate();
 
-nextButton.addEventListener('click', () => {
+const nextButton = NavButton("next", () => {
```

Comment:
Here I just have a question!

Reply:
Sure, we just create a button there and store it in the `const nextButton`

The _NavButton component_ function expects 2 parameters. The first parameter should just be a string and the second one is expected to be a function. The component then uses that for the textContent and the function for the event handler callback.

```
export default function NavButton(**_text_**, **_onClick_**) {
  const button = document.createElement("button");
  button.classList.add("button");
  button.textContent = **text**;
  button.addEventListener("click", **onClick**);
  return button;
}
```

Reply:
Another question we can ask ourselves is: How can the (anonymous) function
```
() => {
    paginationUpdate(1);
}
```
, that we pass down to the component, still correctly call `paginationUpdate` which lives in `index.js`.
I suppose that is because functions retain memory of their outer scopes, where they were defined. That's a concept called closure. Even though we call the anonymous function in the context of the NavButton, it still works. It still has access to the outer scope (global in this case).